### DISCUSSION TITLE
Make Fog/Fog Volume and Volumetric Fog length separate

### DISCUSSION THREAD

**Discussion by DannyDevSe** [godotengine/godot-proposals#9281] [3D]
### Describe the project you are working on

when wanting godrays to get a good resoult you want to decrease the lenght or you will get a seaucure but decreaing the lenght on the volumetric fog also decreases it for the fog so then you cant have fog so would want them to have seperate lenght might be to spesific but i would like it 

### Describe the problem or limitation you are having in your project

chaning max lenght of volumetric fog changes it for fog/fog volumes aswell 

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

it would allow for more freedom in getting the effects you want 

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

n/a

### If this enhancement will not be used often, can it be worked around with a few lines of script?

n/a

### Is there a reason why this should be core and not an add-on in the asset library?

think its good qulity of life but maybe it cant be done

**Comment by Calinou** [comment]
Moving to discussion, as there is no concrete technical implementation provided.

Also, this is not technically feasible. Volumetric fog is rendered in a single pass (with FogVolumes taken into account for this pass), so volumetric fog length must be defined globally and can't be defined on a per-FogVolume basis.

If you only want godrays, a screen-space effect may be more suited depending on the look you're trying to achieve. Godot currently doesn't provide one out of the box, but it should be possible to create one with a custom post-processing shader.

**Reply by yosoyfreeman** [reply]
The problem with god rays made in post processing is that is totally view dependent. You usually want your volumetric light to work together with the fog and the be there regardless of your point of view.
